app/controllers/user_controller.rb

```diff
@@ -0,0 +1,58 @@
+class UserController < ApplicationController
+  before_action :authenticate_user!
+  load_and_authorize_resource
+
+  def index
+    @users = User.where.not(id: current_user.id)
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    @user.admin_id = current_user.id if current_user.role == "Admin"
+    if @user.save
+      flash[:notice] = "Successfully created User."
+      redirect_to user_index_path
+    else
+      flash[:error] = @user.errors.full_messages.join(', ')
+      render action: 'new'
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
```

Comment:
`load_and_authorize_resource` is already loading user and setting `@user`, isn't it?